private/pkg/uuidutil/uuidutil.go

```diff
@@ -29,6 +32,16 @@ func New() (uuid.UUID, error) {
 	return id, nil
 }
 
+// NewULID creates a new ULID for the given timestamp,
+// and returns it in its UUID form.
+func NewULID(timestamp time.Time) (uuid.UUID, error) {
```

Comment:
Why does the caller need to pass a timestamp? Shouldn't this be an implementation detail of the package?

Reply:
It could be, though we generally make it an anti-pattern to call `time.Now()` explicitly in the code, and I think this is an OK interface that makes it easy to test.

Reply:
I am not sure I understand what you mean by anti-pattern, as if we are not calling `time.Now()` here in `uuidutil`, we still need to call it when we use the `NewULID`?

Reply:
I think its worth making exceptions where it makes sense, if a caller needs a random UUID why do they need to bother passing a timestamp. Which opens questions on what _kind_ of time do I need to pass.

Also, can you clarify when we would use `NewULID` vs `NewUUID` ?